Immediately accessing properties on Process objects after Start() can throw NullReferenceException

By calling Process.Start(), then immediately (as in the next line of code) accessing a property on it such as PeakVirtualMemorySize64 can throw a very strange NullReferenceException in the property accessors.  (strange because the text is "error CS0648: '' is a type not supported by the language")
![untitled](https://cloud.githubusercontent.com/assets/7862010/5810953/839237e2-a003-11e4-805f-750d80d1cc31.png)

This is further exacerbated by the lack of (obvious) spin mechanisms on Process class such as the WaitForInputIdle method.  Workarounds are possible but not ideal.  Stepping through in a debugger, the problem will not reproduce.

Noted on : NonpagedSystemMemorySize64
PagedMemorySize64
PagedSystemMemorySize64
PeakPagedMemorySize64
PeakWorkingSet64

--- a/src/System.Diagnostics.Process/src/System/Diagnostics/Process.cs
+++ b/src/System.Diagnostics.Process/src/System/Diagnostics/Process.cs
@@ -314,6 +314,7 @@ namespace System.Diagnostics
         {
             get
             {
+                EnsureState(State.HaveProcessInfo);
                 return _processInfo._poolNonpagedBytes;
             }
         }
@@ -323,6 +324,7 @@ namespace System.Diagnostics
         {
             get
             {
+                EnsureState(State.HaveProcessInfo);
                 return _processInfo._pageFileBytes;
             }
         }
@@ -332,6 +334,7 @@ namespace System.Diagnostics
         {
             get
             {
+                EnsureState(State.HaveProcessInfo);
                 return _processInfo._poolPagedBytes;
             }
         }
@@ -341,6 +344,7 @@ namespace System.Diagnostics
         {
             get
             {
+                EnsureState(State.HaveProcessInfo);
                 return _processInfo._pageFileBytesPeak;
             }
         }
@@ -350,6 +354,7 @@ namespace System.Diagnostics
         {
             get
             {
+                EnsureState(State.HaveProcessInfo);
                 return _processInfo._workingSetPeak;
             }
         }
@@ -359,6 +364,7 @@ namespace System.Diagnostics
         {
             get
             {
+                EnsureState(State.HaveProcessInfo);
                 return _processInfo._virtualBytesPeak;
             }
         }
@@ -424,6 +430,7 @@ namespace System.Diagnostics
         {
             get
             {
+                EnsureState(State.HaveProcessInfo);
                 return _processInfo._privateBytes;
             }
         }
@@ -472,6 +479,7 @@ namespace System.Diagnostics
         {
             get
             {
+                EnsureState(State.HaveProcessInfo);
                 return _processInfo._sessionId;
             }
         }
@@ -533,6 +541,7 @@ namespace System.Diagnostics
         {
             get
             {
+                EnsureState(State.HaveProcessInfo);
                 return _processInfo._virtualBytes;
             }
         }
@@ -643,6 +652,7 @@ namespace System.Diagnostics
         {
             get
             {
+                EnsureState(State.HaveProcessInfo);
                 return _processInfo._workingSet;
             }
         }

